lib/websocket/websocket.js

```diff
@@ -100,12 +111,12 @@ class WebSocket extends EventTarget {
       throw new DOMException('Invalid Sec-WebSocket-Protocol value', 'SyntaxError')
     }
 
-    // 7. Set this's url to urlRecord.
-    this[kWebSocketURL] = urlRecord
+    // 10. Set this's url to urlRecord.
+    this[kWebSocketURL] = new URL(urlRecord.href)
```

Comment:
Why do you need this? This seems unnecessary

Reply:
test/wpt/tests/websockets/Create-http-urls.any.js would fail without this. It's important to note that websocket connections aren't typically opened very much and it's very unlikely that this is a bottleneck if many connections are opened

Reply:
p.s.: I have no idea why the test fails, it took me a little while to track down the issue to here